From: Mohan CM <mohanmanjappa@gmail.com>
commit 193a001e4630da1e67fa217a62d1402c5b1dcc85
Author: Mister Oyster <oysterized@gmail.com>
Date:   Fri Sep 1 11:13:46 2017 +0530

Subject: [PATCH] build: soong: add BOARD_USES_MTK_HARDWARE

Change-Id: Id2c932cb42cae4056eae94e10a6e5c67f91f7348
Signed-off-by: Mister Oyster <oysterized@gmail.com>

diff --git a/android/variable.go b/android/variable.go
index 13456f1..645f592 100644
--- a/android/variable.go
+++ b/android/variable.go
@@ -60,6 +60,12 @@ type variableProperties struct {
 			Cflags []string
 		}
 
+		// Add BOARD_USES_MTK_HARDWARE
+		// This controls Mediatek code specifics
+		Mtk_hardware struct {
+			Cflags []string
+		}		
+		
 		// treble is true when a build is a Treble compliant device.  This is automatically set when
 		// a build is shipped with Android O, but can be overriden.  This controls such things as
 		// the sepolicy split and enabling the Treble linker namespaces.
@@ -133,6 +139,7 @@ type productVariables struct {
 	Eng                        *bool `json:",omitempty"`
 	EnableCFI                  *bool `json:",omitempty"`
 	Treble                     *bool `json:",omitempty"`
+	Mtk_hardware               *bool `json:",omitempty"`
 
 	VendorPath *string `json:",omitempty"`
 
